### PR TITLE
Improve resilience of updates to app's `Order`

### PR DESCRIPTION
**Pull request by luckysori**
Fixes https://github.com/get10101/10101/issues/438.
Fixes https://github.com/get10101/10101/issues/273.

### CODE REVIEW

File: mobile/native/src/trade/order/handler.rs [hunk 1]
```diff
@@ -39,14 +39,22 @@ pub async fn submit_order(order: Order) -> Result<()> {
...
+    let state = OrderState::Filling { execution_price };
+
+    if let Err(e) = update_order_state_in_db_and_ui(order_id, state) {
+        let e_string = format!("{e:#}");
+        match order_failed(Some(order_id), FailureReason::FailedToSetToFilling, e) {
+            Ok(()) => {
+                tracing::debug!(%order_id, "Set order to failed, after failing to set it to filling");
+            }
+            Err(e) => {
+                tracing::error!(%order_id, "Failed to set order to failed, after failing to set it to filling: {e:#}");
```
Comment (bonomat) [inline mobile/native/src/trade/order/handler.rs:53]: Love the log message 😅

File: mobile/native/src/db/mod.rs [hunk 2]
```diff
@@ -35,7 +35,7 @@ pub const MIGRATIONS: EmbeddedMigrations = embed_migrations!();
 /// Sets the number of max connections to the DB.
 ///
 /// This number was arbitrarily chosen and can be adapted if needed.
-const MAX_DB_POOL_SIZE: u32 = 16;
+const MAX_DB_POOL_SIZE: u32 = 1;
```
Comment (bonomat) [inline mobile/native/src/db/mod.rs:38]: Nit: now the value is not arbitrarily chosen anymore. Maybe we can adapt the comment?

File: mobile/native/src/db/models.rs [hunk 3]
```diff
@@ -133,41 +133,70 @@ impl Order {
...
+        let res = conn.exclusive_transaction::<Order, _, _>(|conn| {
+            let order: Order = orders::table
                 .filter(schema::orders::id.eq(order_id.clone()))
-                .set(schema::orders::state.eq(status.0))
-                .execute(conn)?;
+                .first(conn)?;
 
-            if effected_rows == 0 {
-                bail!("Could not update order")
+            tracing::info!("Got here");
```
Comment (holzeis) [inline mobile/native/src/db/models.rs:142]: 🙃 me too 😝

**Review by holzeis — APPROVED** [review]
Awesome! 🚀

File: mobile/native/src/db/models.rs [hunk 4]
```diff
@@ -460,6 +493,33 @@ pub enum OrderState {
...
 }
 
+impl OrderState {
+    /// Determines what state to go to after learning about the latest [`OrderState`] update.
+    ///
+    /// If the state should remain unchanged [`None`] is returned.
+    ///
+    /// TODO: It might be a good idea to introduce a different type that models `OrderUpdates`
+    /// explicitly.
+    fn next_state(&self, latest: Self) -> Option<Self> {
```
Comment (bonomat) [inline mobile/native/src/db/models.rs:503]: What do you think of adding some tests ensuring that this never changes and giving some context about the state transitions?

**Review by bonomat — APPROVED** [review]
Let's give this a try :)

File: mobile/native/src/db/models.rs (hunk 3 above)
Comment (klochowicz) [reply mobile/native/src/db/models.rs:142]: that's why `dbg!` messages are best for debugging  - and clippy can help with removal :p

**Pull request merged**